src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -6551,10 +6551,11 @@ HRESULT DacHeapWalker::Init(CORDB_ADDRESS start, CORDB_ADDRESS end)
                 j++;
             }
         }
-        if ((&g_global_alloc_context)->alloc_ptr != nullptr)
+        gc_alloc_context globalCtx = ((ee_alloc_context)g_global_alloc_context).gc_allocation_context;
```

Comment:
Why do we need the `ee_alloc_context` cast here? 

Also, creating a full copy of the `ee_alloc_context` on stack looks unnecessary - use pointer to `ee_alloc_context` instead?

Reply:
> Why do we need the ee_alloc_context cast here?

g_global_alloc_context is of type __GlobalVal<ee_alloc_context> so dotting directly without a cast wouldn't work.

> use pointer to ee_alloc_context instead?

You could do this:
```
gc_alloc_context* pGlobalCtx = &(&g_global_alloc_context)->gc_allocation_context;
```
Or this:
```
gc_alloc_context* pGlobalCtx = PTR_TO_MEMBER_TADDR(ee_alloc_context, &g_global_alloc_context, gc_allocation_context);
```

Personally, I thought neither of those were as readable and I expect the perf cost in every case is dominated by issuing a remote memory read.

Reply:
Unless I hear that is more important than I realize I'm planning to leave this code as-is.

Reply:
I guess it is ok if it works. It just does not look like typical DACized code.